tests.py

```diff
@@ -90,6 +92,28 @@ def _receive(self):
 
         raise NotImplemented
 
+    def _receive_all(self, signal_to_num):
+        if TestVSM.ipc_module == 'zeromq':
```

Comment:
the zeromq model seems so unsuited to this that I keep thinking if we should not just replace it with a simpler tcp socket module heh